### PR TITLE
reformat the config init output

### PR DESCRIPTION
**Pull request by reidliu41**
Reformat the `config init` output, make it better to view:

```
$ ilab config init
Existing config file was found in:
    /Users/test/.config/instructlab/config.yaml
Do you still want to continue? [y/N]: y

Existing training profiles were found in:
    /Users/test/.local/share/instructlab/internal/train_configuration/profiles
Do you want to restore these profiles to the default values? [y/N]: y

----------------------------------------------------
         Welcome to InstructLab CLI
  This guide will help you to setup your environment
----------------------------------------------------

Please provide the following values to initiate the environment [press Enter for defaults]:
Path to taxonomy repo [/Users/test/.local/share/instructlab/taxonomy]: /Users/test/test-test1

`/Users/test/test-test1` seems to not exist or is empty.
Should I clone https://github.com/instructlab/taxonomy.git for you? [Y/n]: y
Cloning https://github.com/instructlab/taxonomy.git...

Path to your model [/Users/test/.cache/instructlab/models/merlinite-7b-lab-Q4_K_M.gguf]:

Generating config file and profiles:
    /Users/test/.config/instructlab/config.yaml
    /Users/test/.local/share/instructlab/internal/train_configuration/profiles

Please choose a train profile to use.
Train profiles assist with the complexity of configuring InstructLab training for specific GPU hardware.
You can still take advantage of hardware acceleration for training even if your hardware is not listed.
[0] No profile (CPU, Apple Metal, AMD ROCm)
[1] Nvidia A100/H100 x2 (A100_H100_x2.yaml)
[2] Nvidia A100/H100 x4 (A100_H100_x4.yaml)
[3] Nvidia A100/H100 x8 (A100_H100_x8.yaml)
[4] Nvidia L40 x4 (L40_x4.yaml)
[5] Nvidia L40 x8 (L40_x8.yaml)
[6] Nvidia L4 x8 (L4_x8.yaml)
Enter the number of your choice [hit enter for hardware defaults] [0]:
No profile selected - any hardware acceleration for training must be configured manually.

----------------------------------------------------
    Initialization completed successfully!
  You're ready to start using `ilab`. Enjoy!
----------------------------------------------------
```

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Review by nathan-weinberg — APPROVED** [review]

File: src/instructlab/config/init.py
Comment (danmcp) [inline src/instructlab/config/init.py:115]: nit: Include as a \n on the next echo

Same for the other echo()s below

File: src/instructlab/config/init.py
Comment (danmcp) [inline src/instructlab/config/init.py:379]: What happens when the terminal width is smaller than 52?

File: src/instructlab/config/init.py
Comment (reidliu41) [reply src/instructlab/config/init.py:115]: will try later

File: src/instructlab/config/init.py
Comment (reidliu41) [reply src/instructlab/config/init.py:379]: the 52 * "-" is based on the sentence, so if smaller than 52, it will show it in next line. Actually, the long sentence is the same.
image

Comment (danmcp) [reply src/instructlab/config/init.py:379]: The text wrapping is probably fine.  The -'s ideally wouldn't be wider than the window though.

Comment (reidliu41) [reply src/instructlab/config/init.py:379]: is it ok to get the width?
e.g.
```
    terminal_width = shutil.get_terminal_size().columns
    separator = "-" * min(52, terminal_width) 
```

Comment (danmcp) [reply src/instructlab/config/init.py:379]: > is it ok to get the width?

I think so, just needs some testing across environments.

Comment (danmcp) [reply src/instructlab/config/init.py:379]: It would also be good to calc the 52 based on the length of the string.

**Comment by reidliu41** [comment]
hi  team,  I see failed at `e2e-medium-test`, could you please help to take a look?:
```
OSError: Not enough free space to write 67108864 bytes

Epoch 0: 100%|██████████| 11/11 [14:57<00:00, 81.60s/it]
+ rm -rf /tmp/tmp.2xbz7Ttsf3
Error: Process completed with exit code 1.
```

**Comment by danmcp** [comment]
@mergifyio rebase

**Comment by reidliu41** [comment]
thanks

File: src/instructlab/config/init.py
Comment (danmcp) [inline src/instructlab/config/init.py:230]: nit: could move to a \n in the echo below

File: src/instructlab/config/init.py
Comment (danmcp) [inline src/instructlab/config/init.py:335]: nit: could move into a \n in the echo below

File: src/instructlab/config/init.py
Comment (danmcp) [inline src/instructlab/config/init.py:436]: Do we know the cases this exception is happening?  Is there a particular type of exception that's being thrown?  A comment could be helpful here to explain.

File: src/instructlab/config/init.py
Comment (danmcp) [inline src/instructlab/config/init.py:382]: Debating whether this is an improvement.  I would worry the new line will cause some people to miss the context above it.

File: src/instructlab/config/init.py
Comment (danmcp) [inline src/instructlab/config/init.py:374]: nit: could move into a \n in the echo below

File: src/instructlab/config/init.py
Comment (danmcp) [inline src/instructlab/config/init.py:340]: Same comment here that I am not sure whether the newline could cause some people to not make the connection.

File: src/instructlab/config/init.py
Comment (danmcp) [inline src/instructlab/config/init.py:331]: Same comment here that I am not sure whether the newline could cause some people to not make the connection.

File: src/instructlab/config/init.py
Comment (reidliu41) [reply src/instructlab/config/init.py:436]: Actually, I tried to test in `Linux console` or `MacOS terminal` , so far cannot reproduce the error, tried to search, seems might hit `OSError`

File: src/instructlab/config/init.py
Comment (reidliu41) [reply src/instructlab/config/init.py:382]: Personally, compared the old display, it is hard to read, even though show it one by one. Actually, I just try to separate the different `prompt` and `confirm`. if the same prompt, should be together, no new line for it.
```
# ilab config init
Found /root/.config/instructlab/config.yaml, do you still want to continue? [y/N]: y
Existing training profiles were found in /root/.local/share/instructlab/internal/train_configuration/profiles
Do you want to restore these profiles to the default values? [y/N]: y
Welcome to InstructLab CLI. This guide will help you to setup your environment.
Please provide the following values to initiate the environment [press Enter for defaults]:
Path to taxonomy repo [/root/.local/share/instructlab/taxonomy]:
Path to your model [/root/.cache/instructlab/models/merlinite-7b-lab-Q4_K_M.gguf]:

====================================================
-----------------------------------------------------
         Welcome to InstructLab CLI.
  This guide will help you to setup your environment.
-----------------------------------------------------

Please provide the following values to initiate the environment [press Enter for defaults]: <<<
Path to taxonomy repo [/root/.local/share/instructlab/taxonomy]:  <<<
Path to your model [/root/.cache/instructlab/models/merlinite-7b-lab-Q4_K_M.gguf]:  <<<

Generating config file and profiles:
    /root/.config/instructlab/config.yaml
    /root/.local/share/instructlab/internal/train_configuration/profiles
```

File: src/instructlab/config/init.py
Comment (danmcp) [reply src/instructlab/config/init.py:436]: It does look like it would be an OSError.

Maybe just add a comment then explaining why the exception is being caught => Exception can occur when running non interactive scenarios where a terminal isn't associated with the process

File: src/instructlab/config/init.py
Comment (danmcp) [reply src/instructlab/config/init.py:374]: Sorry, not sure I am following.  Would it not work to put a newline before the separator below:

`f"\n{separator}\n`...

Comment (reidliu41) [reply src/instructlab/config/init.py:374]: sorry, sorry, I missed it.  f"\n<<<<<--{separator}\n... is a good position, thanks

**Comment by reidliu41** [comment]
just updated

**Comment by reidliu41** [comment]
hi team, any other ideas? Feel free to share.

**Comment by danmcp** [comment]
Note that I would like another opinion on the formatting questions I had.  Happy to approve or abstain if there is general agreement.

**Comment by JustinXHale** [comment]
UXer here: I think this is a really good improvement on the current view for this. As I installed instructlab I took note of this particular screen because it was jumbled. so really great job. I think i might have some ideas on how to further improve this but not on paper yet. this is great!

File: src/instructlab/config/init.py
Comment (danmcp) [reply src/instructlab/config/init.py:382]: After pulling the code and running init, I no longer have a concern.  The extra whitespace between sections makes it clear you need to read the lines above up to the whitespace.

**Review by danmcp — APPROVED** [review]

**Pull request merged**
